docker/Dockerfile

```diff
@@ -0,0 +1,77 @@
+FROM ubuntu:20.04
+
+# Necessary to install tzdata. It will default to UTC.
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y \
+      python3.9 \
+      # python3.9-distutils is required to install pip
+      # it unfortunately has to install python3.8-minimal as well...
+      python3.9-distutils \
+      # python-is-python3 creates a symlink for python to python3
+      python-is-python3 \
+      # For creating virtual environments
+      python3.9-venv \
+      # wget is used to obtain pip
+      wget \
+      # apache
+      apache2 \
+      apache2-dev \
+      libapr1-dev \
+      apache2-utils \
+      sudo && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set python3 to python3.9 (otherwise, it will be python3.8)
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1
+
+# Install and upgrade pip
+RUN wget -O- https://bootstrap.pypa.io/get-pip.py | python3.9 && \
+    pip install -U pip
```

Comment:
Should we use some extra args to remove cache files since we are inside docker?

Reply:
Looks like adding `--no-cache-dir` keeps it from caching files. I added that.